vm/src/builtins.rs

```diff
@@ -293,9 +293,41 @@ fn builtin_map(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     Ok(vm.ctx.new_list(elements))
 }
 
-// builtin_max
+fn builtin_max(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(
+        vm,
+        args,
+        required = [(x, Some(vm.ctx.int_type())), (y, Some(vm.ctx.int_type()))]
+    );
+
+    use std::cmp::Ordering;
+
+    let order = x.cmp(y);
```

Comment:
Instead of using ordering here, try to use the `__gt__` python method. You can invoke this method by using `vm.call_method(x, "__gt__", vec![y])`. This will then call the python `__gt__` method. Doing this will also allow non-integers to work.

Reply:
When doing that, I get the following error:
```
AttributeError: RefCell { value: [PyObj class "int"] } object has no attribute __gt__
```

Reply:
Yes, this is very good! I means the max function is actually ok, but the int type does not yet implement the `__gt__` method. I think the max function is correct like this.